### PR TITLE
Add type field to error response

### PR DESCRIPTION
**Pull request by viren-nadkarni**
AWS error responses have a field called `Type` which indicates where the blame for the error lies. There is little documentation on this, but HTTP 4xx errors seem to have this field set to `Sender`.

This PR introduces support for this field. To start with, SNS errors are correctly mapped.

The other two templates do not contain the field:
- `SINGLE_ERROR_RESPONSE`: Used for S3 response, see <https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html>
- `ERROR_RESPONSE`: Used for EC2 response, see <https://github.com/boto/botocore/blob/82f6e194716e36c838fc016eddc4a2ef8e449ec4/botocore/parsers.py#L614>

Also see: <https://github.com/boto/botocore/blob/82f6e194716e36c838fc016eddc4a2ef8e449ec4/tests/unit/test_parsers.py#L1200-L1271>

### CODE REVIEW

File: moto/core/exceptions.py [hunk 1]
```diff
@@ -3,8 +3,6 @@
 from typing import Any, List, Tuple, Optional
 import json
 
-# TODO: add "<Type>Sender</Type>" to error responses below?
-
 
 SINGLE_ERROR_RESPONSE = """<?xml version="1.0" encoding="UTF-8"?>
```
Comment (viren-nadkarni) [inline moto/core/exceptions.py:7]: This doesn't seem to contain the `Type` field

<https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html>

File: moto/core/exceptions.py [hunk 2]
```diff
@@ -44,6 +45,9 @@ class RESTError(HTTPException):
     # most APIs use <RequestId>, but some APIs (including EC2, S3) use <RequestID>
     request_id_tag_name = "RequestId"
 
+    # When this field is set, the `Type` field will be included in the response
+    sender_fault = True
```
Comment (bblommers) [inline moto/core/exceptions.py:49]: Minor nitpick - but the attribute-name doesn't make it immediately clear to me what it does. How about `include_type_sender`?

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/core/exceptions.py (hunk 2 above)
Comment (viren-nadkarni) [reply moto/core/exceptions.py]: Changed in 225023df6 :+1:

**Review by bblommers — CHANGES_REQUESTED** [review]
LGTM, just need to get the tests to pass

**Review by bblommers — APPROVED** [review]
Awesome, thanks @viren-nadkarni!

**Pull request merged**